test/test_typechecker.cpp

```diff
@@ -197,3 +197,76 @@ TEST_SUITE("Error positions for unbound parameters")
         CHECK(pos.end != pos.unknown_pos);
     }
 }
+
+TEST_CASE("Ternary operator with clock and double")
+{
+    auto doc = document_fixture{}
+                   .add_global_decl("clock c; double x; void f() { x = true? c : 1.0; }")
```

Comment:
This `true` looks like trivially optimizable.
Instead of `true` use `bool b` argument to the function `f`.
```suggestion
                   .add_global_decl("clock c; double x; void f(bool b) { x = b ? c : 1.0; }")
```

Reply:
This shouldn't matter as optimizations happen at compile time not during parsing. But i'll add it anyway